app/models/concerns/profile_fields.rb

```diff
@@ -62,10 +62,14 @@ def bm_versions
       ).joins(:benchmark).pluck('benchmarks.version')
     end
 
-    def relationships_for(relationship)
+    def relationships_for(rules, rule_groups, relationship)
       RuleGroupRelationship.with_relationships(rules, relationship).or(
         RuleGroupRelationship.with_relationships(rule_groups, relationship)
-      ).each_with_object({}) do |rgr, relationships|
+      )
+    end
+
+    def map_rule_group_relationships(relationship)
+      relationships_for(rules, rule_groups, relationship).each_with_object({}) do |rgr, relationships|
```

Comment:
Is this because of rubocop complaining?

Reply:
Yeah, but also because I wanted to re-use the ```
RuleGroupRelationship.with_relationships(rules, relationship).or(
        RuleGroupRelationship.with_relationships(rule_groups, relationship)
``` part of the `relationships_for` method in `tailoring_valid?` because that was exactly what we needed

Rubocop was complaining a lot when I tried using `RuleGroupRelationship.where(relationship: 'conflicts', left: rules).or(RuleGroupRelationship.where(relationship: 'conflicts', left: groups)).includes(:right).distinct.map(&:right)` in `tailoring_valid?`

Reply:
Okay, I'll need to think about this a bit more...will circle back to this tomorrow.

Reply:
Okay, sounds good 👍  Btw, the rubocop error that I see if I use ```
conflicts = RuleGroupRelationship.where(relationship: 'conflicts', left: rules).or(
      RuleGroupRelationship.where(relationship: 'conflicts', left: rule_groups_for_rules)
    ).distinct.map(&:right)
``` directly in `tailoring_valid?` is 
`Assignment Branch Condition size for tailoring_valid? is too high.`

So that's why I had to move some of it into a different method.